update example app walkthru with new react SDKs (#9)
wix-incubator/wix-cli-docs: guides/example/page_functionality_wix_sdk.md

After setting up permissions, our app can get a list of products from the site's store.

First, install the dependencies we'll need for working with the SDKs:

```
npm install @wix/sdk-react @wix/dashboard-react @wix/stores react-query
```

> In out code, we start by wrapping our exported page component with the [`withDashboard`](TODO: link here) HOC. This will allow us to later use the [`useWixModules`](https://dev.wix.com/docs/sdk/api-reference/sdk-react/hooks#usewixmodules) hook to call Wix APIs.

Reviewer: typo
```suggestion
In our code, we start by wrapping our exported page component with the [`withDashboard`](TODO: link here) HOC. This will allow us to later use the [`useWixModules`](https://dev.wix.com/docs/sdk/api-reference/sdk-react/hooks#usewixmodules) hook to call Wix APIs.
```

> In our code, we start by wrapping our exported page component with the [`withDashboard`](TODO: link here) HOC. This will allow us to later use the [`useWixModules`](https://dev.wix.com/docs/sdk/api-reference/sdk-react/hooks#usewixmodules) hook to call Wix APIs.

Reviewer 2: I think this section should be moved to the previous part of the tutorial where the query wrapper is set up. That way the build flows progressively and people don't have to go back and fix their code. @netanelgilad , if you agree, I can move it over after we deal with the other comments.

Author: I guess that makes sense,  the reason I didn't put it there is that the `withDashboard` doesn't have to do with page design, so didn't want to introduce it without explaining it. If you have a good way to make it "flow", go for it :)

In our previously created `withProvider.tsx` we will wrap our returned component with `withDashboard`:

```diff
import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

> import { WixStyleReactProvider } from '@wix/design-system';
> + import { withDashboard } from "@wix/dashboard-react";

Reviewer: single quotes to be consistent?
```suggestion
import { WixStyleReactProvider } from '@wix/design-system';
+ import { withDashboard } from '@wix/dashboard-react';
```

> + import { withDashboard } from '@wix/dashboard-react';

Reviewer 2: We will have to confirm that this diff markup gets displayed properly in Wix Docs.

export function withProviders(Component: React.ComponentType) {
-  return function () {
+  return withDashboard(function () {
    return (
      <WixStyleReactProvider>
        <QueryClientProvider client={new QueryClient()}>
          <Component />
        </QueryClientProvider>
      </WixStyleReactProvider>
    );
  };
}
```

> Then, in your products page code add the following imports:

Reviewer: We used "ours" so far, not "yours", right?
```suggestion
Then, in our products page code add the following imports:
```

```tsx
import { useWixModules } from '@wix/sdk-react';
import { products } from '@wix/stores';
import { useQuery } from 'react-query';
```

> These imports give us access to the Wix SDK and the particular functionality we need from the SDK.

Reviewer 2: ```suggestion
These imports give us access to the Wix SDK overall and the particular functionality we need.
```

---

> Once imported, we can now intialize the `products` REST APIs module and use the returned `queryProducts` function to access store products. We'll use the `useWixModules` hook in our page component in the following manner:

Reviewer 2: ```suggestion
Once we import everything, we'll use the `useWixModules()` hook create an initialized copy of the `products` module that we can use directly in our page component code. We retrieve the module's [`queryProducts()`](https://dev.wix.com/api/sdk/stores/products/queryproducts) function that we can use to access store products. 
```

```tsx
const { queryProducts } = useWixModules(products);
```

---

> With our SDK set up, we can use it to query products from the site's Wix Store using the [`queryProducts()`](https://dev.wix.com/api/sdk/stores/products/queryproducts) function and handle any errors that might occur.

Reviewer 2: ```suggestion
```

> Here, we query for products whose names start with whatever has been entered into our `AutoComplete` component.

Reviewer 2: ```suggestion
Here, we use the React `useQuery()` hook to query for products whose names start with whatever has been entered into our `AutoComplete` component. We also make sure to handle any errors.
```
